src/input.c

```diff
@@ -1658,50 +1619,15 @@ void game_handle_key_scroll()
 	scrollX = 0;
 	scrollY = 0;
 
-	for (int shortcutId = SHORTCUT_SCROLL_MAP_UP; shortcutId <= SHORTCUT_SCROLL_MAP_RIGHT; shortcutId++) {
-		const int SHIFT = 0x100;
-		const int CTRL = 0x200;
-		const int ALT = 0x400;
-#ifdef __MACOSX__
-		const int CMD = 0x800;
-#endif
-
-		uint16 shortcutKey = gShortcutKeys[shortcutId];
-		uint8 scancode = shortcutKey & 0xFF;
-
-		if (shortcutKey == 0xFFFF) continue;
-		if (!gKeysState[scancode]) continue;
-
-		if (shortcutKey & SHIFT) {
-			if (!gKeysState[SDL_SCANCODE_LSHIFT] && !gKeysState[SDL_SCANCODE_RSHIFT]) continue;
-		}
-		if (shortcutKey & CTRL) {
-			if (!gKeysState[SDL_SCANCODE_LCTRL] && !gKeysState[SDL_SCANCODE_RCTRL]) continue;
-		}
-		if (shortcutKey & ALT) {
-			if (!gKeysState[SDL_SCANCODE_LALT] && !gKeysState[SDL_SCANCODE_RALT]) continue;
-		}
-#ifdef __MACOSX__
-		if (shortcutKey & CMD) {
-			if (!gKeysState[SDL_SCANCODE_LGUI] && !gKeysState[SDL_SCANCODE_RGUI]) continue;
-		}
-#endif
+	if (gKeysHeld[0])
+		scrollY = -1;
+	else if (gKeysHeld[2])
+		scrollY = 1;
 
-		switch (shortcutId) {
-		case SHORTCUT_SCROLL_MAP_UP:
-			scrollY = -1;
-			break;
-		case SHORTCUT_SCROLL_MAP_LEFT:
-			scrollX = -1;
-			break;
-		case SHORTCUT_SCROLL_MAP_DOWN:
-			scrollY = 1;
-			break;
-		case SHORTCUT_SCROLL_MAP_RIGHT:
-			scrollX = 1;
-			break;
-		}
-	}
+	if (gKeysHeld[1])
+		scrollX = -1;
+	else if (gKeysHeld[3])
```

Comment:
If you make it use the last pressed key, then you would have to also make sure it switches back if you let go of that last pressed key.